variantlib/commands/make_variant.py

```diff
@@ -0,0 +1,222 @@
+            "times as needed"
+        ),
+    )
+
+    parser.add_argument(
+        "-f",
+        "--file",
+        dest="input_filepath",
+        type=pathlib.Path,
+        required=True,
+        help="Wheel file to process",
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output_directory",
+        type=pathlib.Path,
+        required=True,
+        help="Output Directory to use to store the Wheel Variant",
+    )
+
+    parsed_args = parser.parse_args(args)
+
+    wheel_file_re = re.compile(
+        r"""^(?P<namever>(?P<name>[^\s-]+?)-(?P<ver>[^\s-]*?))
+        ((-(?P<build>\d[^-]*?))?-(?P<pyver>[^\s-]+?)-(?P<abi>[^\s-]+?)-(?P<plat>[^\s-]+?)
+        \.whl|\.dist-info)$""",
+        re.VERBOSE,
+    )
+
+    input_filepath: pathlib.Path = parsed_args.input_filepath
+    output_directory: pathlib.Path = parsed_args.output_directory
+
+    # Input Validation
+    if not input_filepath.is_file():
+        raise FileExistsError(f"Input Wheel File `{input_filepath}` does not exists.")
+
+    if not output_directory.is_dir():
+        raise FileExistsError(f"Output Directory `{output_directory}` does not exists.")
+
+    # Input Validation - Wheel Filename is valid and non variant already.
+    wheel_info = wheel_file_re.match(input_filepath.name)
+    if not wheel_info:
+        raise ValueError(f"`{input_filepath.name}` is not a valid wheel filename.")
+
+    # Transform properties into a VariantDescription
+    properties_args: list[str] | None = parsed_args.properties
+    if properties_args is None or len(properties_args) == 0:
+        raise ValueError("At least one Variant Property is required.")
+
+    vprops = [VariantProperty.from_str(vprop) for vprop in properties_args]
+    vdesc = VariantDescription(properties=vprops)
+
+    with tempfile.TemporaryDirectory() as _tmpdir:
+        tempdir = pathlib.Path(_tmpdir)
+        wheel_unpack(input_filepath, tempdir)
+
+        dirname = input_filepath.name.split("-py3")[0]
+        wheel_dir = tempdir / dirname
+
+        if not wheel_dir.exists():
+            raise FileNotFoundError(wheel_dir)
+
+        distinfo_dir = wheel_dir / f"{dirname}.dist-info"
+
+        if not distinfo_dir.exists():
+            raise FileNotFoundError(distinfo_dir)
+
+        metadata_f = distinfo_dir / "METADATA"
+
+        if not metadata_f.exists():
+            raise FileNotFoundError(metadata_f)
+
+        with metadata_f.open(mode="r+") as file:
+            # Read all lines
+            lines = file.readlines()
+
+            # Remove trailing empty lines
+            while lines and lines[-1].strip() == "":
+                lines.pop()
+
+            # Move the file pointer to the beginning
+            file.seek(0)
+
+            # Write back the non-empty content
+            file.writelines(lines)
+
+            # Truncate the file to remove any remaining old content
+            file.truncate()
+
+            file.write(f"Variant-hash: {vdesc.hexdigest}\n")
```

Comment:
This seems to assume that the metadata will never have longdescription. Also, altering metadata should really be done via the email message class (or `packaging.metadata`, but that will fail on new keys).